### PR TITLE
Update rust toolchain to 1.80.0

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
<https://blog.rust-lang.org/2024/07/25/Rust-1.80.0.html>

### CODE REVIEW

**Comment by FirelightFlagboy** [comment]
Depends on #7805

File: libparsec/crates/types/src/id.rs [hunk 1]
```diff
@@ -646,15 +646,19 @@ crate::impl_from_maybe!(Option<HumanHandle>);
...
+/// This should not be confused with `RealmRole` (which represents the different roles
+/// a user can have in a realm) !
+///
+/// The different possible profiles are:
+/// - `Admin` can invite and revoke users and on top of what standard user can do.
+/// - `Standard` user can create new realms and invite new devices for himself.
+/// - `Outsider` is only able to collaborate on existing realm and can only
+///   access redacted certificates (i.e. the realms created by an outsider
+///   cannot be shared and the outsider cannot be OWNER/MANAGER
+///   on a realm shared with him)
```
Comment (FirelightFlagboy) [inline libparsec/crates/types/src/id.rs:660]: I would have keep the description of each variant under it's own doc comment

Comment (touilleMan) [reply libparsec/crates/types/src/id.rs]: done.

**Review by touilleMan — APPROVED** [review]
currently Rust 1.80 is causing some issues with the Windows build, so we choose to stick to our current version of Rust while also merge the changes requested by Clippy (so future upgrade will be simpler)

**Pull request merged**